### PR TITLE
Generic Hardware Wallet Errors

### PR DESCRIPTION
**Pull request by blurpesec**
[ch6374]

### Description
Introduces better error messaging for hardware wallet unlock errors.

##### Changes
- Refactored from scss to styled-components for Hardware WalletUnlock and SignTransaction components.
- Introduced error handling for hardware interactions.

##### Screenshots
Selection_024

### CODE REVIEW

**Review by vilsbole — CHANGES_REQUESTED** [review]
Love the scss deletions!

**Comment by GuillaumeRx** [comment]
Can we center the error text ?

File: src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -113,7 +115,9 @@ const svgIcons = {
   whitepaper,
   experience,
   'ledger-icon': ledgerIcon,
+  'ledger-icon-lg': ledgerLgIcon,
   'trezor-icon': trezorIcon,
+  'trezor-icon-lg': trezorLgIcon,
```
[the remark this answers is not in the document]
Comment (blurpesec) [reply src/components/Icon/Icon.tsx:120]: They're different.

File: src/components/SignTransactionWallets/Hardware.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+import { DEFAULT_NETWORK } from '@config';
+import { fAccounts } from '@fixtures';
+import { translateRaw } from '@translations';
+import { WalletId } from '@types';
+
+import { SignTxHardwareUI } from './Hardware';
+
+export default { title: 'SignTransaction' };
```
[the remark this answers is not in the document]
Comment (blurpesec) [reply src/components/SignTransactionWallets/Hardware.stories.tsx:10]: Can't see this in storybook unless i have exports on these.

File: src/components/SignTransactionWallets/Hardware.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  signerDescription: translateRaw('SIGN_TX_LEDGER_DESCRIPTION', {
+    $network: DEFAULT_NETWORK
+  }),
+  helpCopy: 'LEDGER_HELP',
+  referralCopy: 'LEDGER_REFERRAL',
+  senderAccount: { ...fAccounts[0], wallet: WalletId.LEDGER_NANO_S_NEW },
+  isTxSignatureRequestDenied: true
+};
+
+export const HardwareWalletUI = () => {
```
[the remark this answers is not in the document]
Comment (blurpesec) [reply src/components/SignTransactionWallets/Hardware.stories.tsx:23]: reference https://github.com/MyCryptoHQ/MyCrypto/pull/3668#discussion_r552198847

File: src/components/WalletUnlock/Ledger.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+import { fNetwork } from '@fixtures';
+import { noOp } from '@utils';
+
+import { LedgerUnlockUI } from './Ledger';
+
+export default { title: 'WalletUnlock/LedgerNanoS' };
```
[the remark this answers is not in the document]
Comment (blurpesec) [reply src/components/WalletUnlock/Ledger.stories.tsx:8]: reference https://github.com/MyCryptoHQ/MyCrypto/pull/3668#discussion_r552198847

File: src/components/WalletUnlock/Ledger.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    completed: false,
+    error: {
+      code: '0x1111',
+      message: 'A looooooong error message used to test the output of ledger connections.'
+    }
+  },
+  handleNullConnect: noOp
+};
+
+export const LedgerPanel = () => {
```
[the remark this answers is not in the document]
Comment (blurpesec) [reply src/components/WalletUnlock/Ledger.stories.tsx:31]: reference https://github.com/MyCryptoHQ/MyCrypto/pull/3668#discussion_r552198847

File: src/components/WalletUnlock/NewTrezor.tsx [hunk 6]
```diff
@@ -2,9 +2,10 @@ import React, { useState } from 'react';
```
[the remark this answers is not in the document]
Comment (blurpesec) [reply src/components/WalletUnlock/NewTrezor.tsx:2]: consolidated the UI portions

File: src/components/WalletUnlock/NewTrezor.tsx [hunk 7]
```diff
@@ -84,55 +145,63 @@ const TrezorDecrypt = ({ formData, onUnlock }: OwnProps) => {
...
+    <Heading fontSize="32px" textAlign="center" fontWeight="bold">
+      {translate('UNLOCK_WALLET')}{' '}
+      {translateRaw('YOUR_WALLET_TYPE', { $walletType: translateRaw('X_TREZOR') })}
+    </Heading>
+    <TrezorPanel>
+      <TrezorContent>
+        <TrezorDescription>
+          {translate('TREZOR_TIP')}
+          <TrezorImageContainer>
+            <img src={ConnectTrezor} />
```
[the remark this answers is not in the document]
Comment (blurpesec) [reply src/components/WalletUnlock/NewTrezor.tsx:168]: Yeh.

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**